docs/configuration/nsis.md

```diff
@@ -37,31 +37,36 @@ Two options are available — [include](#NsisOptions-include) and [script](#Nsis
 Keep in mind — if you customize NSIS script, you should always state about it in the issue reports. And don't expect that your issue will be resolved.
 
 1. Add file `build/installer.nsh`.
-2. Define wanted macro to customise: `customHeader`, `preInit`, `customInit`, `customUnInit`, `customInstall`, `customUnInstall`, `customRemoveFiles`, `customInstallMode`.
-    
+2. Define wanted macro to customise: `customHeader`, `preInit`, `customInit`, `customUnInit`, `customInstall`, `customUnInstall`, `customRemoveFiles`, `customInstallMode`, `customWelcomePage`.
+
     !!! example
         ```nsis
         !macro customHeader
           !system "echo '' > ${BUILD_RESOURCES_DIR}/customHeader"
         !macroend
-        
+
         !macro preInit
           ; This macro is inserted at the beginning of the NSIS .OnInit callback
           !system "echo '' > ${BUILD_RESOURCES_DIR}/preInit"
         !macroend
-        
+
         !macro customInit
           !system "echo '' > ${BUILD_RESOURCES_DIR}/customInit"
         !macroend
-        
+
         !macro customInstall
           !system "echo '' > ${BUILD_RESOURCES_DIR}/customInstall"
         !macroend
-        
+
         !macro customInstallMode
-          # set $isForceMachineInstall or $isForceCurrentInstall 
+          # set $isForceMachineInstall or $isForceCurrentInstall
           # to enforce one or the other modes.
         !macroend
+
+        !macro customWelcomePage
+          # Welcome Page is not added by default for installer.
+          !insertMacro MUI_PAGE_WELCOME
+        !macroend
```

Comment:
Hiya! I'm not sure I understand this change.  What is `MUI_PAGE_WELCOME` and why does the developer have to custom specify it in a separate include.nsh file?

I only see `customWelcomePage` being used here:
https://github.com/electron-userland/electron-builder/blob/28cb86bdcb6dd0b10e75a69ccd34ece6cca1d204/packages/app-builder-lib/templates/nsis/assistedInstaller.nsh#L7-L11
So I don't understand how adding this `MUI_PAGE_WELCOME` to the `customWelcomePage` will work for an installer. Could you clarify this change?

Reply:
Hi @mmaietta my understanding after going through Nullsoft Scriptable Install System docs is that one will need to specify the pages manually and the macro `customWelcomePage` introduced in this PR https://github.com/electron-userland/electron-builder/pull/2384 allows dev to display the Welcome Page when running the setup for installation.

Without modifying the logic of Electron Builder for generating NSIS build, the `build/installer.nsh` file as suggested by the docs didn't mentioned anything about  `customWelcomePage` macro. Specifying the `!insertmacro MUI_PAGE_WELCOME` in the `customWelcomePage` macro allows me to restore the Welcome Page.

As for what is `MUI_PAGE_WELCOME`? It is one of the available page macros (more specifically it refers to the installer page) as part of NSIS Modern User Interface https://nsis.sourceforge.io/Docs/Modern%20UI/Readme.html (scroll down to Pages section). If you see in the `assistedInstaller.nsh` file, there is `MUI_UNPAGE_WELCOME` and `MUI_UNPAGE_INSTFILES` for the uninstaller setup.

https://github.com/electron-userland/electron-builder/blob/28cb86bdcb6dd0b10e75a69ccd34ece6cca1d204/packages/app-builder-lib/templates/nsis/assistedInstaller.nsh#L67-L74

Let me know if you need real world examples. I've been searching for how to insert the Welcome Page for the installer setup and there is not many results except for this issue https://github.com/electron-userland/electron-builder/issues/3809 although I found it after figuring out the `customWelcomePage` macro method. Ironically the uninstaller setup does have "Welcome Page" on line 68 in the code snippet above.

Reply:
Thanks for the explanation! I'm not familiar with NSIS scripting so that helps a lot. I'm kicking off the final CI build/tests now